app/src/main/java/org/sil/hearthis/MainActivity.java

```diff
@@ -35,16 +36,101 @@ public void onClick(View view) {
 				LaunchSyncActivity();
 			}
 		});
+		
+		if (requestRecordAudioPermission()) {
+			// We already have permission to record audio, proceed normally.
+			launchChooseBookIfProject();
+		}
+		// Otherwise the function will have requested permission, which launches the dialog as
+		// an activity; we must not launch another activity which would suppress the dialog.
+		// If we eventually get permission, we will go ahead with launchChooseBookIfProject().
+	}
+
+	// completely arbitrary, especially when we're only asking for one dangerous permission.
+	// I just thought it might be useful to have a fairly distinctive number, for debugging.
+	private final int MY_PERMISSIONS_REQUEST_RECORD_AUDIO = 17;
 
-		launchChooseBookIfProject();
+	// Although the app declares that it needs permission to record audio, because it is considered
+    // a dangerous permission the user must grant it explicitly through this procedure from API23 on.
+    // In theory, the user could revoke it at any time, and we should check it every time we need it.
+    // Since our app may be resumed when the user thinks it is being started up, it would be tempting
+	// to check on resume; but this is problematic because a resume happens when the user closes the
+	// permission dialog. If we then ask again if it is still denied, we are refusing to take "no"
+	// for an answer, and displaying an infinite succession of permission dialogs.
+	// In practice, I think users will understand that HT needs permission to record audio and will
+	// grant it the first time the program starts up after installation. At least they will get that
+	// one request instead of just being left to wonder why it won't record (more often, they wonder
+	// why it won't play back, because it isn't obvious that it didn't record). So this may not be
+	// the best or final solution to this problem, but it's a big and very possibly sufficient step.
+	private boolean requestRecordAudioPermission() {
+		if (ContextCompat.checkSelfPermission(this,
+				Manifest.permission.RECORD_AUDIO)
+				!= PackageManager.PERMISSION_GRANTED) {
+
+			// Permission is not granted
+			// Should we show an explanation? This test returns true if the user has previously
```

Comment:
I agree that the rationale seems obvious (to us), but based on what I read in this article, it might be worth giving an explanation: https://www.nngroup.com/articles/permission-requests/

Reply:
It actually wasn't obvious, so much so that I initially removed this code after adding the code to request it when the user touches the record button. But in fact, we'd like record permission immediately, so we can show the sound level indicator. I've added an explanatory dialog.